### ISSUE TITLE
BatchTileMapRenderer beginRender() and endRender() when ownsRender is false

### ISSUE THREAD

**Issue by nunoflores** [libgdx/libgdx#2991]
Regarding BatchTileMapRenderer, when the Batch is passed onto the constructor, thus the renderer not owning the batch, it should not call the batch.begin() and batch.end() on beginRender() and endRender() methods (should it?). I've created a subclass that overrides both methods to change this and its compatible with other batch drawing. 

@Override
	protected void beginRender() {
		if (ownsBatch) {
			super.beginRender();
		}
	}
	
	@Override
	protected void endRender() {
		if (ownsBatch) {
			super.endRender();
		}
	}

**Comment by MobiDevelop** [comment]
The idea is that you would override the begin/end render methods if you
don't want to have the default behavior of beginning and ending the batch.
I can see where you are coming from and would be ok with changing this if
others agree.

**Comment by azurlake** [comment]
I have subclassed BatchTiledMapRenderer as well. In my case, I found unnecessary the call to batch.setProjectionMatrix (...) inside the setView (camera) method, causing a reflush of the batch, unnecessary if the batch has already set its projection matrix - as far as I understand. The default behavior is ok though, so I think it is just ok to subclass it.

**Comment by GameplayJDK** [comment]
I would agree.. To me it was an unexpected behavior because I assumed that the renderer required the batch to be begun before rendering, since it didn't own it. I was forced to subclass (in my case) `OrthogonalTiledMapRenderer` and add in the `if (ownsBatch) { ... }` statements by hand.

It seems redundant (to me) to be able to pass a batch if you have no real control over it...

It might be an a good idea to add such a subclass out of the box.
